src/language-server/universal-data-definition-language-formatter.ts

```diff
@@ -128,11 +162,123 @@ export class UniversalDataDefinitionLanguageFormatter extends AbstractFormatter
         this.formatObj(elem);   
     }
 
+    protected formatLogicalEnumerated(lenum: ast.LogicalEnumerated): void {
+        this.formatContainer(lenum)
+        lenum.label.forEach(label => {
+            this.formatContainer(label)
```

Comment:
The label is an array of LogicalEnumeratedSet | LogicalEnumerationLabel which are both containers

Reply:
It might be a LogicalEnumeratedSet, which is a container, or a LogicalEnumerationLabel, which is not a container. In these circumstances, remember that the top level `format` method will figure out the type of label - so all you need to do is call `format`, which will then call the appropriate formatter for either LogicalEnumerationLabel or LogicalEnumeratedSet. Your current approach will not handle multiple levels of nesting appropriately.